### ISSUE TITLE
[Test] Stylize Video (with mask)

### ISSUE THREAD

**Issue by s9roll7** [s9roll7/animatediff-cli-prompt-travel#111]
<div><video controls src="https://github.com/s9roll7/animatediff-cli-prompt-travel/assets/118420657/0b7ced31-1165-46b2-a354-c73a4cc39417" muted="false"></video></div>
<br>

<div><video controls src="https://github.com/s9roll7/animatediff-cli-prompt-travel/assets/118420657/6801d88f-cd1c-4080-98ae-7c8f7233df3c" muted="false"></video></div>
<br>

<div><video controls src="https://github.com/s9roll7/animatediff-cli-prompt-travel/assets/118420657/4e3b5d5f-0f91-4025-a5e7-5596b7038890" muted="false"></video></div>
<br>

**Comment by Jasonlin00** [comment]
can you provide the full prompt.json?please

**Comment by TomatoSlasher** [comment]
is there a way to directly mask the latents instead of post-processing, i think it would lead to much better results @s9roll7

**Comment by s9roll7** [comment]
Do you mean inpaint?
That sounds interesting too.

**Comment by TomatoSlasher** [comment]
yes, with inpainting it's much better to only generate the masked area without the need to generate the whole frame and cut out the foreground with masks which leads to unnecessary detail being generated like in the examples above, i'm trying to implement it for this repo but i'm still facing some issues, i'd love to talk to you about it @s9roll7.

**Comment by s9roll7** [comment]
I think each method has its advantages.
The advantage of the method I implemented this time is that we can change the shape of the conversion target

**Comment by jonaaathan** [comment]
hi @s9roll7 the videos are said to be corrupted. Mind re uploading the video ? thanks

**Comment by s9roll7** [comment]
> hi @s9roll7 the videos are said to be corrupted. Mind re uploading the video ? thanks

I have no problems in my environment.

**Comment by jonaaathan** [comment]
you right. works on another one's computer but not mine's 2 browser. Thanks anyway!

**Comment by alanhuang67** [comment]
HELLO! I have some question about  Stylize Video (with mask)

1. When I generate the background images, the image size changed (I saw Prompt changes also) is it correct or I need to modify by my selg?
2. can I just change background image without changing front ground?  how should I setup?

thanks!

**Comment by alanhuang67** [comment]
> I'm a little unclear on the situation.
> Do you mean that some areas of the picture are missing?
> Can you upload the image that went wrong?
My orgional sitting for prompt.json:
            "width": 512,
            "height": 336,
            "length": 180,
            "context": 16,
            "overlap": 4,
            "stride": 0

and the setting in bg_2023-10-19T19-28-01 become
            "width": 448,
            "height": 568,
            "length": 180,
            "context": 16,
            "overlap": 4,
            "stride": 0
which the width and high changed, I thought the fd may change the width and height but not sure if bg will change or not?

**Comment by s9roll7** [comment]
I see
I'm resizing it by default to save time.
If you want to improve quality, change to a larger size.

**Comment by s9roll7** [comment]
> I see I'm resizing it by default to save time. If you want to improve quality, change to a larger size.

Since the aspect ratio of the background does not change, there may be no point in adjusting the size.
I will change the background size so that it does not change in the next update

**Comment by hylarucoder** [comment]
HI, @s9roll7 , 

After applying the "stylize composite" effect, the result appears green. Should I adjust some parameters to control the amount of green?


Origin Image from result

![00000000](https://github.com/s9roll7/animatediff-cli-prompt-travel/assets/5625783/6a41f40e-7154-4e0d-b511-abb7a4cdb96b)

`cp_2023-10-20T08-49-58/fg_00_2023-10-20T08-49-58/00000000.png`

![00000000](https://github.com/s9roll7/animatediff-cli-prompt-travel/assets/5625783/f9696496-36bf-464c-bbc3-876d431dc16d)


`cp_2023-10-20T08-49-58/bg_00_2023-10-20T08-49-58/00000000.png`

![00000000](https://github.com/s9roll7/animatediff-cli-prompt-travel/assets/5625783/e541a0eb-06af-4733-ba37-1f02b31d125b)

**Comment by s9roll7** [comment]
I think it is the influence of the composition method.
The green color of its legs comes from the color of the trees in the background.

**Comment by hylarucoder** [comment]
@s9roll7 Is there a way to minimize the environmental impact?

I tried modifying the compose method with a simplified OpenCV version (code bellow). The color of the leg looks good, but the edges of the mask appear fragile. 

I need to adjust the LaplacianPyramidBlender, but which parameters can I set?

Is there a better way to enhance composite color and mask edge quality?

**Thank you very much.**

```python
        bg = Image.open(bg)
        fg = Image.fromarray(fg_array)
        # Create mask
        fg_pixels = fg.load()
        width, height = fg.size
        mask = Image.new('L', fg.size)
        mask_pixels = mask.load()

        for y in range(height):
            for x in range(width):
                r, g, b = fg_pixels[x, y]
                if g > r and g > b:  # Simple green detection
                    mask_pixels[x, y] = 0
                else:
                    mask_pixels[x, y] = 255

        # Blend images
        result = Image.composite(fg, bg, mask)

        # Save result
        result.save(save_path)
```

![00000000](https://github.com/s9roll7/animatediff-cli-prompt-travel/assets/5625783/139f2dd5-af7f-454f-8455-cdba1b0ea87a)
